docs/source/algorithms.rst

```diff
@@ -0,0 +1,180 @@
+Run quantum algorithms
+======================
+
+Perceval provides a processor-centric syntax to run an algorithm locally or remotely, on a simulator or an actual QPU.
+
+Build a processor
+------------------
+
+A :ref:`Processor` is a composite element aiming at simulating an actual QPU, in real world conditions.
+
+* It holds a single photon :ref:`Source`
+* It is a composition of unitary circuits and non-unitary components
+* Input and output ports can be defined, with encoding semantics
+* Logical post-processing can be set-up through heralded modes (ancillas) and a final post-selection function
+* It contains the means of simulating the setup it describes with one of :ref:`The Backends` which are provided
+
+Create a processor from scratch
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As an example, let's create locally a heralded CNOT gate from its circuit:
+
+>>> import perceval as pcvl
+>>> from perceval.components import BS, PERM, Port, Encoding
+>>> c_hcnot = (pcvl.Circuit(8, name="Heralded CNOT")
+...            .add((0, 1, 2), PERM([1, 2, 0]))
+...            .add((4, 5), BS.H())
+...            .add((5, 6, 7), PERM([1, 2, 0]))
+...            .add((3, 4), BS.H())
+...            .add((2, 3), BS.H(theta=self.theta1, phi_bl=np.pi, phi_tr=np.pi/2, phi_tl=-np.pi/2))
+...            .add((4, 5), BS.H(theta=self.theta1))
+...            .add((3, 4), BS.H())
+...            .add((5, 6, 7), PERM([2, 1, 0]))
+...            .add((1, 2), PERM([1, 0]))
+...            .add((2, 3), BS.H(theta=self.theta2))
+...            .add((4, 5), BS.H(theta=self.theta2, phi_bl=np.pi, phi_tr=np.pi/2, phi_tl=-np.pi/2))
+...            .add((5, 6), PERM([1, 0]))
+...            .add((4, 5), BS.H())
+...            .add((4, 5), PERM([1, 0]))
+...            .add((0, 1, 2), PERM([2, 1, 0])))
+>>> processor_hcnot = pcvl.Processor("SLOS", c_hcnot)
+>>> processor_hcnot.add_herald(0, 0)
+...                .add_herald(1, 1)
+...                .add_port(2, Port(Encoding.DUAL_RAIL, 'data'))
+...                .add_port(4, Port(Encoding.DUAL_RAIL, 'ctrl'))
+...                .add_herald(6, 0)
+...                .add_herald(7, 1)
+>>> pcvl.pdisplay(processor_hcnot, recursive=False)
+
+.. figure:: _static/img/heralded-cnot-processor.png
+    :align: center
+
+    Heralded CNOT gate rendering - heralded modes are not shown for readability
+
+Processor composition
+^^^^^^^^^^^^^^^^^^^^^
+
+Processors can also be composed with one another. That's for example, how :ref:`Qiskit converter` outputs a complex
```

Comment:
* drop the *also*
* "one another" => "another processor"
* "That's" => "That is"
* "how Quiskit Converter" => "how the Qiskit converter"